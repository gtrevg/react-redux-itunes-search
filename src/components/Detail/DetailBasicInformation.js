import React from 'react';

const DetailBasicInformation = () => {
	return <div>Basic Information</div>;
};

export default DetailBasicInformation;
