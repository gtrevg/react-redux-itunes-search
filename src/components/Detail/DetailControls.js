import React from 'react';

const DetailControls = props => {
	const { preview } = props;
	if (!preview) return null;
	return (
		<div>
			<audio src={preview} controls>
				Sorry, your browser does not support the audio element.
			</audio>
		</div>
	);
};

export default DetailControls;
