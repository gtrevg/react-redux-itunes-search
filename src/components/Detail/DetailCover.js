import React from 'react';

import './DetailCover.css';

const DetailCover = props => {
	const { cover } = props;
	if (!cover) return null;
	//Trick for improving the size
	const cover300px = cover.replace('100x100bb', '300x300bb');
	return <img alt="Cover not found" src={cover300px} className="DetailCover" />;
};

export default DetailCover;
