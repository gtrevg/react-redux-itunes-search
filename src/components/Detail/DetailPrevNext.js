import React from 'react';
import { Tooltip, Button, Icon } from 'antd';

const lookUpIdIndex = (id, dataSource) => {
	return dataSource.findIndex(item => item.key === id);
};

const getPreviousItem = (idIndex, dataSource) => {
	if (idIndex === 0) return null;
	return dataSource[idIndex - 1];
};

const getNextItem = (idIndex, dataSource) => {
	if (idIndex === dataSource.lenght - 1) return null;
	return dataSource[idIndex + 1];
};

const DetailPrevNext = props => {
	const { id, dataSource } = props;
	const idIndex = lookUpIdIndex(id, dataSource);
	if (idIndex >= 0) {
		const previousItem = getPreviousItem(idIndex, dataSource);
		const nextItem = getNextItem(idIndex, dataSource);
		return (
			<Button.Group size="large">
				<Tooltip
					placement="bottomRight"
					title={`${previousItem.title} by ${previousItem.artist}`}
				>
					<Button
						type="primary"
						style={{ width: '120px', margin: '10px' }}
						href={`/detail?id=${previousItem.key}`}
					>
						<Icon type="left" />
						Next
					</Button>
				</Tooltip>
				<Tooltip
					placement="bottomLeft"
					title={`${nextItem.title} by ${nextItem.artist}`}
				>
					<Button
						type="primary"
						style={{ width: '120px', margin: '10px' }}
						href={`/detail?id=${nextItem.key}`}
					>
						Previous
						<Icon type="right" />
					</Button>
				</Tooltip>
			</Button.Group>
		);
	} else {
		return null;
	}
};

export default DetailPrevNext;
