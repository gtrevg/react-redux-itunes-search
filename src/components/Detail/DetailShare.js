import React from 'react';

const DetailShare = () => {
	return <div>Share</div>;
};

export default DetailShare;
