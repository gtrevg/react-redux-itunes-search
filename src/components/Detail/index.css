.Detail {
	margin: 10px;
}
