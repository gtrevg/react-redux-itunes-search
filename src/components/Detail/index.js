import React from 'react';
import { Link } from 'react-router-dom';

const Detail = () => {
	return (
		<div>
			<Link to="/">Return to search</Link>
		</div>
	);
};

export default Detail;
