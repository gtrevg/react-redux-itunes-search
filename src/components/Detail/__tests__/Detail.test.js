import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Detail from '../index';

describe('Detail component', () => {
	const initialEntries = ['/', '/detail'];
	const component = (
		<MemoryRouter initialEntries={initialEntries}>
			<Detail />
		</MemoryRouter>
	);

	it('Renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(component, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Snapshot matchs', () => {
		const tree = renderer.create(component).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
