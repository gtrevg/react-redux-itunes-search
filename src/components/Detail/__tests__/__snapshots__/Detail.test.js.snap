// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail component Snapshot matchs 1`] = `
<mocked
  name="Antd Spin"
  props={
    Object {
      "size": "large",
      "spinning": false,
      "wrapperClassName": "Detail",
    }
  }
>
  <mocked
    name="DetailBasicInformation"
    props={
      Object {
        "artist": "Michael Jackson",
        "title": "Billie Jean",
      }
    }
  />
  <mocked
    name="DetailCover"
    props={
      Object {
        "cover": "http://is2.mzstatic.com/image/thumb/Music127/v4/8a/65/be/8a65bef2-f23d-e43d-9124-f5e4293513f7/source/100x100bb.jpg",
      }
    }
  />
  <mocked
    name="DetailControls"
    props={
      Object {
        "preview": "https://audio-ssl.itunes.apple.com/apple-assets-us-std-000001/AudioPreview122/v4/c6/50/11/c6501132-e865-3711-175a-ddb79114e42f/mzaf_3806132797788612279.plus.aac.p.m4a",
      }
    }
  />
  <mocked
    name="DetailPrevNext"
    props={
      Object {
        "search": Object {
          "dataSource": Array [
            Object {
              "album": "Bad",
              "artist": "Michael Jackson",
              "cover": "http://is2.mzstatic.com/image/thumb/Music/v4/05/c4/1c/05c41ca0-b614-a91c-e63e-2fd9a59d19e8/source/60x60bb.jpg",
              "genre": "Pop",
              "key": 559334764,
              "lenght": 257766,
              "price": "1.29 USD",
              "release": "1987-08-31T07:00:00Z",
              "title": "Smooth Criminal",
            },
            Object {
              "album": "Thriller",
              "artist": "Michael Jackson",
              "cover": "http://is4.mzstatic.com/image/thumb/Music/v4/ec/2b/17/ec2b170d-2ecb-50af-02eb-39f0decc2847/source/60x60bb.jpg",
              "genre": "Pop",
              "key": 269573364,
              "lenght": 294188,
              "price": "1.29 USD",
              "release": "1982-11-30T08:00:00Z",
              "title": "Billie Jean",
            },
            Object {
              "album": "The Essential Michael Jackson",
              "artist": "Michael Jackson",
              "cover": "http://is2.mzstatic.com/image/thumb/Music127/v4/8a/65/be/8a65bef2-f23d-e43d-9124-f5e4293513f7/source/60x60bb.jpg",
              "genre": "Pop",
              "key": 159294478,
              "lenght": 320905,
              "price": "1.29 USD",
              "release": "1987-08-31T07:00:00Z",
              "title": "Man In the Mirror",
            },
          ],
          "isLoading": false,
        },
      }
    }
  />
  <mocked
    name="DetailShare"
    props={
      Object {
        "artist": "Michael Jackson",
        "title": "Billie Jean",
      }
    }
  />
</mocked>
`;
